scarb/src/manifest_editor/add.rs

```diff
@@ -52,9 +52,20 @@ impl Op for AddDependency {
 
         let dep = Dep::resolve(*self, ctx)?;
 
+        let was_sorted = tab.as_table_like().map_or(true, |dep_table| {
+            let values = dep_table.get_values();
+            is_sorted(values.iter().map(|(key, _)| key[0]))
```

Comment:
How does this deal with this input? Doesn't this panic?

```table
[dependencies]
foo.bar = "1.0.0"
```

Yes, that's still invalid code, but this **may** become valid with #13. No need to deal with this specially, only I'd like to ensure it's resilient to panics

Reply:
Seem like this has more to do with how the manifest file is being deserialized? specifically the `PackageName` struct

Reply:
Yes, but take into account that here we are not deserializing it